### PR TITLE
Fix the Dockerfile for missing PHP module & Rewrite the compare script to be smarter

### PR DESCRIPTION
**Pull request by roland-d**
As the title says, I updated the Dockerfile to account for a missing PHP package . The compare script has been modified to be smarter in comparing the files and when to run composer and npm.

I have tested this all locally and is working fine now.

### CODE REVIEW

File: compare.sh [hunk 1]
```diff
@@ -148,15 +202,33 @@ else
...
+if ! [ -d upload ]; then
+    # Reference cache not mounted
+    echo "Upload folder does not exist, creating it"
+    mkdir upload
+fi
+
 # Move files to upload directory
-mkdir upload
-mv "$CMP_BUILD_DIR/$CMP_ARCHIVE.zip" ./upload
-echo "Finished zipping build."
```
Comment (HLeithner) [inline compare.sh:159]: Sure you want to remove the link to the download position?

File: compare.sh [hunk 2]
```diff
@@ -5,66 +5,123 @@
...
+echo "#####################"
+echo "Checkout pull request"
+echo "#####################"
+
+cd $CMP_SLAVE_FOLDER
 echo "Current directory: "$(pwd)
 
+# Download the pull request diff and apply it
+echo "Apply pull request $DRONE_PULL_REQUEST"
+curl -L https://github.com/joomla/joomla-cms/pull/"$DRONE_PULL_REQUEST".diff | git apply --binary --reject 2> apply.log
```
Comment (HLeithner) [inline compare.sh:84]: Why are you apply this by hand? your current directory has diff already applied

File: compare.sh (hunk 1 above)
Comment (roland-d) [reply compare.sh:159]: This is not removed, I fixed the variabel name. See the two lines below here.

**Comment by roland-d** [comment]
@HLeithner What do we need to do to make this work? Is there anywhere I can test the script on drone?

**Comment by HLeithner** [comment]
Do you mean this script or the reference point the user need to install so we can build a diff that the patchtester can install?

**Comment by roland-d** [comment]
@HLeithner I just pushed my change that the diff is build against the latest nightly build. With this I think we are good to go.

**Pull request merged**